core-common/src/main/java/com/example/core_common/BaseViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.core_common
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.core_data.domain.entity.BaseDataItem
+import io.reactivex.Single
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+abstract class BaseViewModel<T: BaseDataItem> : ViewModel() {
+
+    val dataLiveData: MutableLiveData<List<T>> = MutableLiveData<List<T>>()
+    val isLoadingLiveData = MutableLiveData<Boolean>()
+    val errorLiveData = MutableLiveData<Unit>()
+    var dataList: List<T> = emptyList()
+    private val disposable: CompositeDisposable = CompositeDisposable()
+
+    override fun onCleared() {
+        super.onCleared()
+        disposable.clear()
+    }
+
+    fun requestData(useCase: Single<List<T>>) {
+        isLoadingLiveData.postValue(true)
+        disposable.add(
+            useCase.subscribeOn(Schedulers.io())
+                .doAfterTerminate { isLoadingLiveData.postValue(false) }
+                .subscribe(
+                    { value ->
+                        dataLiveData.postValue(value)
+                        dataList = value
+                    },
+                    { error ->
+                        errorLiveData.postValue(Unit)
+                        error.printStackTrace()
+                    }
+                )
+        )
+    }
+
+    fun searchMarket(searchValue: String) {
+        dataLiveData.postValue(dataList.filter {
+            it.itemName.contains(
+                searchValue,
+                ignoreCase = true
+            )
+        })
```

Comment:
Текущий вариант очень сложно читается. Как тебе такой?
```suggestion
        val filteredData = dataList.filter { item ->
            item.itemName.contains(searchValue, ignoreCase = true)
        }

        dataLiveData.postValue(filteredData)
```